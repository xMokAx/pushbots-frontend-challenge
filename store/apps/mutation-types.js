export const GET_APPS_START = 'GET_APPS_START'
export const GET_APPS_SUCCESS = 'GET_APPS_SUCCESS'
export const GET_APPS_FAILURE = 'GET_APPS_FAILURE'
