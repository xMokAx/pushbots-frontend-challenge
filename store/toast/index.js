import { SHOW_TOAST, CLOSE_TOAST } from './mutation-types'

const icons = {
  error: 'mdi-alert',
  success: 'mdi-check',
  info: 'mdi-information',
  warning: 'mdi-exclamation'
}

export const state = () => ({
  isToastShown: false,
  toastOptions: {
    color: '',
    text: '',
    icon: ''
  }
})

export const mutations = {
  [SHOW_TOAST]: (state, toastOptions) => {
    state.isToastShown = true
    state.toastOptions = {
      icon: icons[toastOptions.color],
      ...toastOptions
    }
  },
  [CLOSE_TOAST]: (state) => {
    state.isToastShown = false
  }
}

export const actions = {}

export const getters = {
  toastOptions: (state) => state.toastOptions
}
