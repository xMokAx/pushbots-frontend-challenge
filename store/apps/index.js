import { SHOW_TOAST } from '../toast/mutation-types'
import {
  GET_APPS_START,
  GET_APPS_SUCCESS,
  GET_APPS_FAILURE
} from './mutation-types'

export const state = () => ({
  isLoading: false,
  totalApps: 0,
  limit: 4,
  skip: 0,
  page: 1,
  apps: []
})

export const mutations = {
  [GET_APPS_START]: (state, page) => {
    state.isLoading = true
    state.page = page
    state.skip = page * state.limit - state.limit
  },
  [GET_APPS_SUCCESS]: (state, data) => {
    state.isLoading = false
    state.apps = data.data
    state.totalApps = data.total
  },
  [GET_APPS_FAILURE]: (state) => {
    state.isLoading = false
  }
}

export const actions = {
  async fetchApps({ commit, state }, page) {
    commit(GET_APPS_START, page)
    const { skip, limit } = state
    try {
      const res = await this.$axios.get(
        `/api/apps?take=${limit}&skip=${skip}&sortBy=title&direction=desc`
      )
      commit(GET_APPS_SUCCESS, res.data)
    } catch (error) {
      commit(
        `toast/${SHOW_TOAST}`,
        {
          color: 'error',
          text: 'Failed to get apps. Please, check your connection.'
        },
        { root: true }
      )
      commit(GET_APPS_FAILURE)
    }
  }
}
