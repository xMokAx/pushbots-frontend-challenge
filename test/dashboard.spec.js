import { mount, createLocalVue } from '@vue/test-utils'
import Vue from 'vue'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import { state, mutations, actions } from '../store/apps'
import DashboardPage from '../pages/dashboard.vue'

const vuetify = new Vuetify()
Vue.use(Vuetify)

const localVue = createLocalVue()
localVue.use(Vuex)

const storeOptions = {
  modules: {
    apps: {
      namespaced: true,
      state: state(),
      mutations,
      actions
    }
  }
}
const store = new Vuex.Store(storeOptions)

describe('Dashboard Page', () => {
  it('Should match snapshot', () => {
    const wrapper = mount(DashboardPage, {
      localVue,
      vuetify,
      store,
      stubs: ['nuxt-link']
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
