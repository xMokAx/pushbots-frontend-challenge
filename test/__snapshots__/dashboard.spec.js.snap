// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Page Should match snapshot 1`] = `
"<main class=\\"container flex-grow-1\\"><div class=\\"v-tabs pb-3 theme--light\\"><div role=\\"tablist\\" class=\\"v-item-group theme--light v-slide-group v-tabs-bar v-tabs-bar--is-mobile v-tabs-bar--show-arrows primary--text grey lighten-5\\"><div class=\\"v-slide-group__prev v-slide-group__prev--disabled\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate v-icon--disabled mdi mdi-chevron-left theme--light\\"></i></div><div class=\\"v-slide-group__wrapper\\"><div class=\\"v-slide-group__content v-tabs-bar__content\\"><div class=\\"v-tabs-slider-wrapper\\" style=\\"transition: none;\\"><div class=\\"v-tabs-slider\\"></div></div> <button tabindex=\\"0\\" aria-selected=\\"true\\" role=\\"tab\\" class=\\"v-tab v-tab--active\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-check theme--light\\"></i>
      complete
    </button><button tabindex=\\"0\\" aria-selected=\\"false\\" role=\\"tab\\" class=\\"v-tab\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-settings theme--light\\"></i>
      in setup
    </button><button tabindex=\\"0\\" aria-selected=\\"false\\" role=\\"tab\\" class=\\"v-tab\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-share theme--light\\"></i>
      shared
    </button><button tabindex=\\"0\\" aria-selected=\\"false\\" role=\\"tab\\" class=\\"v-tab\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-chart-bar theme--light\\"></i>
      active
    </button></div></div></div></div> <div class=\\"row\\"></div> <ul class=\\"py-4 px-4 v-pagination theme--light\\" page=\\"1\\"><li><button type=\\"button\\" class=\\"v-pagination__navigation v-pagination__navigation--disabled\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-chevron-left theme--light\\"></i></button></li><li><button type=\\"button\\" class=\\"v-pagination__navigation v-pagination__navigation--disabled\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-chevron-right theme--light\\"></i></button></li></ul> <div class=\\"v-overlay theme--dark\\" style=\\"z-index: 5;\\"><div class=\\"v-overlay__scrim\\" style=\\"opacity: 0; background-color: rgba(255, 255, 255, 0.7);\\"></div></div></main>"
`;
